docs/groovy-script/mods/mwc/crafting_station.md

```diff
@@ -0,0 +1,207 @@
+---
+title: "MWC Crafting"
+titleTemplate: "Modern Warfare Cubed | CleanroomMC"
+description: "Converts multiple items into another item, with a way to recover some of original resources when dismantled. recipes with group ``BULLET, MAGAZINE, GRENADE`` can be only crafted via Ammo Press, while recipes with ``GUN, ATTACHMENT_NORMAL, ATTACHMENT_MODIFICATION`` categories can only be crafted with Crafting Station."
+source_code_link: "https://github.com/Cubed-Development/Modern-Warfare-Cubed/blob/next/src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java"
+---
+
+# MWC Crafting (Modern Warfare Cubed)
+
+## Description
+
+Converts multiple items into another item, with a way to recover some of original resources when dismantled. recipes with group ``BULLET, MAGAZINE, GRENADE`` can be only crafted via Ammo Press, while recipes with ``GUN, ATTACHMENT_NORMAL, ATTACHMENT_MODIFICATION`` categories can only be crafted with Crafting Station.
+
+## Identifier
+
+Refer to this via any of the following:
+
+```groovy:no-line-numbers {1}
+mods.mwc.crafting_station/* Used as page default */ // [!code focus]
+mods.mwc.craftingstation
+mods.mwc.craftingStation
+mods.mwc.CraftingStation
+mods.mwc.ammo_press
+mods.mwc.ammopress
+mods.mwc.ammoPress
+mods.mwc.AmmoPress
+```
+
+
+## Adding Recipes
+
+### Recipe Builder
+
+Just like other recipe types, the MWC Crafting also uses a recipe builder.
+
+Don't know what a builder is? Check [the builder info page](../../getting_started/builder.md) out.
+
+:::::::::: details mods.mwc.crafting_station.recipeBuilder() {open id="abstract"}
+- `IngredientList<IIngredient>`. Sets the item inputs of the recipe. Requires less than or equal to 27.
+
+    ```groovy:no-line-numbers
+    input(IIngredient...)
+    input(IIngredient, double)
+    input(double, IIngredient...)
```

Comment:
be consistent about the order - have `double` be first, or dont allow varargs. might want to ditch the `input` list entirely and store `CraftingEntry` directly (with a method that is `x(double d, IIngredient i) { return x(new CraftingEntry(i.toMcIngredient(), i.getAmount(), d)); }`).